naff/api/events/processors/reaction_events.py

```diff
@@ -34,6 +35,7 @@ async def _handle_message_reaction_change(self, event: "RawGatewayEvent", add: b
                         message.reactions.pop(i)
                     else:
                         message.reactions[i] = r
+                    reaction = r
                     break
             else:
                 message.reactions.append(
```

Comment:
You need to store the reaction variable here too. Easiest fix would be 

```python
reaction = Reaction.from_dict(
    {
        "count": 1,
        "me": author.id == self.user.id,  # type: ignore
        "emoji": emoji.to_dict(),
        "message_id": message.id,
        "channel_id": message._channel_id,
    },
    self,  # type: ignore
)
message.reactions.append(reaction)
```

Reply:
Thanks, missed it